### PR TITLE
fix: addpair should prevent adding same base and quote assets (#1559)

### LINKED ISSUE #1559: Same currency pair can be added
### Steps to reproduce

1. Add Pair BTC/BTC
2. Place order to BTC/BTC order book and fill it

### Expected behaviour

No opportunity to add <symbol>/<same_symbol> because:
1. its useless action
2. for now eth/eth swap leads to SwapTimeOut and 500 Connext Error

### Actual behaviour

Pair Added, Swap performed
Note: These Swaps work only for lndbtc and lndltc and dont work for Connext pairs (leads to SwapTimeOut and 500 Connext Error).

### PR DESCRIPTION
**Pull request by rsercano**
resolves #1559 
image

### CODE REVIEW

**Review by raladev — DISMISSED** [review]

File: lib/orderbook/errors.ts [hunk 1]
```diff
@@ -75,6 +76,10 @@ const errors = {
     message: `order with local id ${localId} has a quantity of ${holdQuantity} satoshis on hold, try again later`,
     code: errorCodes.QUANTITY_DOES_NOT_MATCH,
   }),
+  PAIR_CURRENCIES_CANT_BE_SAME: (baseCurrency: string, quoteCurrency: string) => ({
```
Comment (erkarl) [inline lib/orderbook/errors.ts:79]: Not sure about the naming of this error. At the same time I don't have a better alternative. Thoughts @sangaman?

**Review by erkarl — COMMENTED** [review]
Changes look good. Just 1 minor comment/question regarding the error name that we might want to address.

File: lib/orderbook/errors.ts (hunk 1 above)
Comment (sangaman) [reply lib/orderbook/errors.ts:79]: Yeah it's a bit clunky, how about `DUPLICATE_PAIR_CURRENCIES`?

**Review by sangaman — CHANGES_REQUESTED** [review]
We should add this new error code to `getGrpcError.ts` as one of the `INVALID_ARGUMENT` cases.

**Comment by rsercano** [comment]
Thanks @erkarl @sangaman, fixed

**Review by raladev — APPROVED** [review]

**Review by sangaman — APPROVED** [review]

**Pull request merged**